include/onnxruntime/core/framework/tensor.h

```diff
@@ -58,15 +58,18 @@ using BufferNakedPtr = void*;
 class Tensor final {
  public:
   /**
-     Create tensor with given type, shape, pre-allocate memory and allocator info.
-  */
-  Tensor(MLDataType p_type,
-         const TensorShape& shape,
-         BufferNakedPtr p_data,
-         const OrtAllocatorInfo& alloc,
-         AllocatorPtr deleter = nullptr,
+   * Create tensor with given type, shape, pre-allocate memory and allocator info.
+   * This function won't check if the preallocated buffer(p_data) has enough room for the shape.
+   */
```

Comment:
Explicitly say Tensor does not own the data and will not delete it. It's somewhat implied by taking a raw pointer, but as it's a non-const pointer it's better to be explicit.